src/components/CommentForm.js

```diff
@@ -0,0 +1,73 @@
+import { useState, useContext } from "react";
+import { postComment } from "../api";
+import { UserContext } from "../context/userContext";
+import styles from "./CommentForm.module.css";
+
+const CommentForm = ({ review_id, setNewComment }) => {
+  const [commentText, setCommentText] = useState("");
+  const { user } = useContext(UserContext);
+  const [posted, setPosted] = useState(false);
+
+  const handleSubmit = (event) => {
+    setPosted(true);
+    event.preventDefault();
+    postComment(review_id, user.username, commentText).then(() => {
+      setNewComment((current) => current + 1);

```

Comment:
Instead of having the setNewComment after posting we could do it before to optimistically render the comment :)